extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/dataportabilityproject/transfer/microsoft/calendar/MicrosoftCalendarExporter.java

```diff
@@ -0,0 +1,162 @@
+            }
+        } catch (IOException e) {
+            e.printStackTrace();  // FIXME log error
+            return new ExportResult<>(ExportResult.ResultType.ERROR, "Error retrieving calendar: " + e.getMessage());
+        }
+
+        List<CalendarEventModel> calendarEventModels = new ArrayList<>();
+
+        for (CalendarModel calendarModel : calendarModels) {
+            String id = calendarModel.getId();
+            Request.Builder eventsBuilder = getBuilder(calculateEventsUrl(id), authData);
+
+            try (Response graphResponse = client.newCall(eventsBuilder.build()).execute()) {
+                ResponseBody body = graphResponse.body();
+                if (body == null) {
+                    return new ExportResult<>(ExportResult.ResultType.ERROR, "Error retrieving calendar: response body was null");
+                }
+                String graphBody = new String(body.bytes());
+                Map graphMap = objectMapper.reader().forType(Map.class).readValue(graphBody);
+
+                // String nextLink = (String) graphMap.get(ODATA_NEXT);
+                // ContinuationData continuationData = nextLink == null ? null : new ContinuationData(new GraphPagination(nextLink));
+
+                @SuppressWarnings("unchecked") List<Map<String, Object>> rawEvents = (List<Map<String, Object>>) graphMap.get("value");
+                if (rawEvents == null) {
+                    return new ExportResult<>(ExportResult.ResultType.END);
+                }
+
+                for (Map<String, Object> rawEvent : rawEvents) {
+                    Map<String, String> properties = new HashMap<>();
+                    properties.put(CALENDAR_ID, id);
+                    TransformResult<CalendarEventModel> result = transformerService.transform(CalendarEventModel.class, rawEvent, properties);
+                    if (result.hasProblems()) {
+                        // discard
+                        // FIXME log problem
+                        continue;
+                    }
+                    calendarEventModels.add(result.getTransformed());
+                }
+
+            } catch (IOException e) {
+                e.printStackTrace();  // FIXME log error
+                return new ExportResult<>(ExportResult.ResultType.ERROR, "Error retrieving contacts: " + e.getMessage());
+            }
+        }
+
+
+        CalendarContainerResource resource = new CalendarContainerResource(calendarModels, calendarEventModels);
+        return new ExportResult<>(ExportResult.ResultType.END, resource, null);
+    }
+
+    private String calculateEventsUrl(String eventId) {
+        return baseUrl + String.format(EVENTS_URL, eventId);
+    }
+
+    @Override
+    public ExportResult<CalendarContainerResource> export(TokenAuthData authData, ExportInformation exportInformation) {
+        throw new UnsupportedOperationException();
+    }
+
+    private Request.Builder getBuilder(String url, TokenAuthData authData) {
```

Comment:
private static

Reply:
Do you think that should be static? It's a private method only called by other non-static members of that class

Reply:
It may be helpful to readers of the method to indicate that it doesn't depend on the state of member variables in the class versus trying to discern from reading the contents.

Reply:
Sure we can do this

Reply:
Not sure if you wanted to update it now or not.